functions/go/set-image/transformer/images_transformer.go

```diff
@@ -0,0 +1,170 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/custom"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/image_util"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+// SetImage supports the set-image workflow, it uses Config to parse functionConfig, Transform to change the image
+type SetImage struct {
+	// Image is the desired image
+	Image image_util.Image `json:"image,omitempty" yaml:"image,omitempty"`
+	// ConfigMap keeps the data field that holds image information
+	ConfigMap map[string]string `json:"data,omitempty" yaml:"data,omitempty"`
+	// ONLY for kustomize, AdditionalImageFields is the user supplied fieldspec
+	AdditionalImageFields image_util.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+	// context logs each detailed result
+	context *fn.Context
```

Comment:
nit: let's remove this attribute. You can pass in the context to each function that need to log results.